clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -146,6 +146,7 @@ public synchronized void assign(Collection<TopicPartition> partitions) {
         ensureNotClosed();
         committed.clear();
         this.subscriptions.assignFromUser(new HashSet<>(partitions));
+        this.paused.retainAll(partitions);
```

Comment:
Piggy-backed fix, when unassigned those partitions, they should also be removed from paused list.